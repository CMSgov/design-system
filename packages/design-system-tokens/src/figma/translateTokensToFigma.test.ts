import testGetResponse from './test/test-get-response';
import {
  generatePostVariablesPayload,
  getCollectionsByName,
  getVariablesByCollection,
} from './translateTokensToFigma';

describe('getCollectionsByName', () => {
  it('returns an object of collections keyed by name', () => {
    const collections = getCollectionsByName(testGetResponse);
    expect(collections).toHaveProperty('System');
    expect(collections).toHaveProperty('Theme');
  });
});

describe('getVariablesByCollection', () => {
  it('returns a nested object organizing variables into collections', () => {
    const variables = getVariablesByCollection(testGetResponse);
    const system = variables['VariableCollectionId:111'];
    const theme = variables['VariableCollectionId:222'];
    expect(system['color/ocean/50']).toMatchSnapshot();
    expect(theme['theme/color/primary']).toMatchSnapshot();
  });
});

// describe('generatePostVariablesPayload', () => {
//   it('matches snapshot', () => {
//     const payload = generatePostVariablesPayload(, testGetResponse);

//   });
// })
