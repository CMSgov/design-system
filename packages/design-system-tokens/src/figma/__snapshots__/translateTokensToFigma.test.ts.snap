// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getVariablesByCollection returns a nested object organizing variables into collections 1`] = `
Object {
  "codeSyntax": Object {},
  "description": "",
  "hiddenFromPublishing": false,
  "id": "oce050",
  "key": "color/ocean/50",
  "name": "color/ocean/50",
  "remote": false,
  "resolvedType": "COLOR",
  "scopes": Array [
    "ALL_SCOPES",
  ],
  "valuesByMode": Object {
    "ModeId:1": Object {
      "a": 1,
      "b": 0.9725490212440491,
      "g": 0.9450980424880981,
      "r": 0.9019607901573181,
    },
  },
  "variableCollectionId": "VariableCollectionId:111",
}
`;

exports[`getVariablesByCollection returns a nested object organizing variables into collections 2`] = `
Object {
  "codeSyntax": Object {},
  "description": "",
  "hiddenFromPublishing": false,
  "id": "pri300",
  "key": "theme/color/primary",
  "name": "theme/color/primary",
  "remote": false,
  "resolvedType": "COLOR",
  "scopes": Array [
    "ALL_SCOPES",
  ],
  "valuesByMode": Object {
    "ModeId:21": Object {
      "id": "oce500",
      "type": "VARIABLE_ALIAS",
    },
    "ModeId:22": Object {
      "id": "oce500",
      "type": "VARIABLE_ALIAS",
    },
    "ModeId:23": Object {
      "id": "oce500",
      "type": "VARIABLE_ALIAS",
    },
    "ModeId:24": Object {
      "id": "oce500",
      "type": "VARIABLE_ALIAS",
    },
  },
  "variableCollectionId": "VariableCollectionId:222",
}
`;
